drools-wb-screens/drools-wb-guided-dtree-editor/drools-wb-guided-dtree-editor-client/src/main/java/org/drools/workbench/screens/guided/dtree/client/widget/GuidedDecisionTreeWidget.java

```diff
@@ -416,163 +424,169 @@ protected BaseGuidedDecisionTreeShape getParentNode( final BaseGuidedDecisionTre
         }
-        hint.animate( AnimationTweener.LINEAR,
-                      new AnimationProperties(),
-                      ANIMATION_DURATION,
-                      new IAnimationCallback() {
-
-                          @Override
-                          public void onStart( final IAnimation iAnimation,
-                                               final IAnimationHandle iAnimationHandle ) {
-                              //Nothing to do
-                          }
-
-                          @Override
-                          public void onFrame( final IAnimation iAnimation,
-                                               final IAnimationHandle iAnimationHandle ) {
-                              //Lienzo's IAnimation.getPercent() passes values > 1.0
-                              final double pct = iAnimation.getPercent() > 1.0 ? 1.0 : iAnimation.getPercent();
-                              hint.setAlpha( 1 - pct );
-                              hint.getLayer().batch();
-                          }
-
-                          @Override
-                          public void onClose( final IAnimation iAnimation,
-                                               final IAnimationHandle iAnimationHandle ) {
-                              canvasLayer.remove( hint );
-                              isGettingStartedHintVisible = false;
-                          }
-                      } );
+        hint.animate(AnimationTweener.LINEAR,
+                     new AnimationProperties(),
+                     ANIMATION_DURATION,
+                     new IAnimationCallback() {
+
+                         @Override
+                         public void onStart(final IAnimation iAnimation,
+                                             final IAnimationHandle iAnimationHandle) {
+                             //Nothing to do
+                         }
+
+                         @Override
+                         public void onFrame(final IAnimation iAnimation,
+                                             final IAnimationHandle iAnimationHandle) {
+                             //Lienzo's IAnimation.getPercent() passes values > 1.0
+                             final double pct = iAnimation.getPercent() > 1.0 ? 1.0 : iAnimation.getPercent();
+                             hint.setAlpha(1 - pct);
+                             hint.getLayer().batch();
+                         }
+
+                         @Override
+                         public void onClose(final IAnimation iAnimation,
+                                             final IAnimationHandle iAnimationHandle) {
+                             canvasLayer.remove(hint);
+                             isGettingStartedHintVisible = false;
+                         }
+                     });
+    }
 
+    void setUiRoot(final WiresBaseTreeNode uiRoot) {
+        this.uiRoot = uiRoot;
     }
 
+    void setModel(final GuidedDecisionTree model) {
+        this.model = model;
+    }
```

Comment:
I know that these `setUiRoot` and `setModel` was added because of tests, just interested if you tried to use the original one [1] in the test. I haven't tried locally if possible, just asking.

[1]
 ```public void setModel(final GuidedDecisionTree model,
                         final boolean isReadOnly)```

Reply:
@jomarko I tried that route first.. but it opened a huge can of worms of trying to mock a bunch of Lienzo internals; as that ```setModel(model, isReadOnly)``` method calls ```super.clear()``` that tries to draw the ```Canvas```. In the end I added the methods you saw to make my life (much) easier.